src/compiler/transformers/declarations.ts

```diff
@@ -707,13 +712,16 @@ export function transformDeclarations(context: TransformationContext) {
         return newParam;
     }
 
-    function shouldPrintWithInitializer(node: Node) {
-        return canHaveLiteralInitializer(node) && resolver.isLiteralConstDeclaration(getParseTreeNode(node) as CanHaveLiteralInitializer); // TODO: Make safe
+    function shouldPrintWithInitializer(node: Node): node is Node & { initializer: PrimitiveLiteral } {
+        if(canHaveLiteralInitializer(node) && node.initializer && !node.type && isPrimitiveLiteralValue(node.initializer)) {
+            return isDeclarationReadonly(node) || isVariableDeclaration(node) && isVarConstLike(node);
+        }
+        return false;
     }
 
     function ensureNoInitializer(node: CanHaveLiteralInitializer) {
         if (shouldPrintWithInitializer(node)) {
-            return resolver.createLiteralConstValue(getParseTreeNode(node) as CanHaveLiteralInitializer, symbolTracker); // TODO: Make safe
+            return clonePrimitiveLiteralValue(node.initializer)

```

Comment:
We clone the primitive in order to remove any source specific features. Such as `_` or hex or octal or usage of `` ` `` or `'` with strings. We could just use the original node, but if there is a new way to write a primitive literal it might not be backward compatible with older versions of TypeScript. If we don't think that is a problem, I would prefer to use the original node with its original format.